### PR TITLE
Add the directory structure for RFC documents

### PR DESCRIPTION
**Pull request by akukanov**
The proposed RFC structure and process generally match those recently accepted by oneTBB (see oneapi-src/oneTBB#1464) with some modifications, primarily just in the description. To see only the modifications, exclude the first commit when viewing the changes.

### CODE REVIEW

File: rfcs/experimental/README.md
Comment (danhoeflinger) [inline rfcs/experimental/README.md:14]:
```suggestion
should include a list of the exit conditions that need to be met to move the feature from
experimental to fully supported. These conditions might include demonstrated
```

File: rfcs/README.md
Comment (danhoeflinger) [inline rfcs/README.md:28]: Does it make sense to have either a subdirectory within "Supported" named "Extension", or another top level directory for "Extension"?  

I'm wondering about features which might be extensions either because they might rely upon hardware / compiler specific functionality or perhaps features which we want to show have more stability / robustness but are are ready for the specification for some reason. We've discussed labelling extensions in the past and it may make sense to codify that with some structure.

The time limit on experimental got me thinking about this.  Both ranges and async APIs could've fit into such a category (until recently in the case of Ranges API).  They have been reasonably stable and even depended on internally from non-experimental code, but were still experimental.

**Review by danhoeflinger — COMMENTED** [review]
This generally looks good.  One small wording suggestion, and a suggestion for an "extension" category / directory.

File: rfcs/README.md
Comment (dcbenito) [inline rfcs/README.md:5]: Please add periods to the end of each bullet point. In combination with the initial sentence, they are all complete sentences (unlike the list below with a single state).

File: rfcs/README.md
Comment (dcbenito) [inline rfcs/README.md:32]: Please remove "yet"; it is redundant.

File: rfcs/README.md
Comment (dcbenito) [inline rfcs/README.md:69]: Please add commas:
Once**,** at least**,** two maintainers approve the PR, it is merged into the main branch
as an RFC proposed for implementation.

File: rfcs/experimental/README.md
Comment (dcbenito) [inline rfcs/experimental/README.md:23]: Please change "It" to "They" to keep continuity from the previous sentence. Or change "Proposals" to "A proposal".

File: rfcs/experimental/README.md
Comment (dcbenito) [inline rfcs/experimental/README.md:26]: I think this should be more declarative: 
Proposals should not stay in the experimental folder for longer than a year.

File: rfcs/supported/README.md
Comment (dcbenito) [inline rfcs/supported/README.md:9]: Please add commas:
The RFC document should**,** in that case**,** have a link
to the formal wording in the specification.

File: rfcs/template.md
Comment (dcbenito) [inline rfcs/template.md:8]: Please add an apostrophe to "user's", it's possessive in this case, or change to "user"

File: rfcs/template.md
Comment (dcbenito) [inline rfcs/template.md:37]: Please capitalize "does".

File: rfcs/template.md
Comment (dcbenito) [inline rfcs/template.md:46]: Please capitalize "some".

File: rfcs/template.md
Comment (dcbenito) [inline rfcs/template.md:48]: Please capitalize "the".

**Review by dcbenito — CHANGES_REQUESTED** [review]
Minor changes requested. Thanks.

File: rfcs/README.md
Comment (akukanov) [reply rfcs/README.md:28]: I prefer to wait for a precedent or maybe two, and decide based on that experience.

Comment (danhoeflinger) [reply rfcs/README.md:28]: Sure, I understand why we may want this only in rare cases and not advertise it. This will avoid the temptation of using it as an escape valve for experimental projects which stall rather than pushing for graduation to the specification.

I'm OK with leaving it out until we see it is necessary.

**Review by dcbenito — DISMISSED** [review]
LGTM

**Review by danhoeflinger — DISMISSED** [review]

**Review by mmichel11 — COMMENTED** [review]
The RFC process looks clear and good to me. I have just left two small comments on wording / grammar.

**Review by dmitriy-sobolev — COMMENTED** [review]
I've left a couple of minor suggestions. Overall, the documents look very good.

File: rfcs/README.md
Comment (akukanov) [reply rfcs/README.md:40]: The specification is not needed for an extension.

However I do not think that "either... or" instead of "both... and" makes anything more explicit. A **single** fully supported feature can be **_either_** specified **_or_** an extension; but **a set** fully supported features consists of **_both_** specified features **_and_** extensions. Here I am speaking about the plurality of features.

File: rfcs/README.md
Comment (akukanov) [reply rfcs/README.md:43]: Similar to the distinction about specified features and extensions that we discussed with Dan in the comments, I prefer to wait for real cases and not introduce extra process steps upfront.

**Review by dmitriy-sobolev — APPROVED** [review]

**Review by mmichel11 — APPROVED** [review]

**Pull request merged**
